components/script/dom/gpubuffer.rs

```diff
@@ -47,11 +48,12 @@ pub enum GPUBufferState {
 #[derive(JSTraceable, MallocSizeOf)]
 pub struct GPUBufferMapInfo {
     #[ignore_malloc_size_of = "Rc"]
-    pub mapping: Rc<RefCell<Vec<u8>>>,
+    #[no_trace]
+    pub mapping: Arc<Mutex<Vec<u8>>>,
```

Comment:
Let's add a doc that explains that this is an arc because the spidermonkey integration requires it to be thread-safe.